src/snowflake/snowpark/async_job.py

```diff
@@ -331,6 +333,15 @@ def result(
             _AsyncResultType(result_type.lower()) if result_type else self._result_type
         )
         self._cursor.get_results_from_sfqid(self.query_id)
+        if self._num_statements is not None:
+            for _ in range(self._num_statements - 1):
+                self._cursor.nextset()
+            # TODO: Once we support fetch_pandas_all for multi-statement query in connector, we could remove this
+            #   workaround.
+            self._cursor.execute(
+                f"select * from table(result_scan('{self._cursor.sfqid}'))"
+            )
+
```

Comment:
I don't understand this section. we execute a result scan but whom do we share the results with for the intermediate queries.

Reply:
Today we cannot do `fetch_pandas_all` for multi-statement query because we need arrow result format, but the result of intermediate queries are child queries of a SQL stored proc (the backend support of multi-statement query), and SQL stored proc does not support arrow for child queries.

As far as snowpark is concerned, we only need the result of the last statement of a multi-statement query. And this is why we skip the results until the last one. Let me add more comment here.

Reply:
That said, we probably only need to do `result_scan` when we need pandas. Let me optimize that as well.

Reply:
Is this the only way to skip intermediate queries and get to the result of the last one? out of scope for this PR but we can add a way to skip to last result in python connector itself.

Reply:
Skipping to the last query are handled by 

```
            for _ in range(self._num_statements - 1):
                self._cursor.nextset()
```

The `result_scan` here is to make sure that we get the result in ARROW format. It can be applied to any of the statements of this query.

It does sound like a good idea to have a quick way to get the last result :)